### PR TITLE
feat(workflow): Add workflow for generating bill-of-materials

### LINKED ISSUE #466: Generate bill of materials with GitHub workflow
**Is your feature request related to a problem? Please describe.**
Customers interested in monaco should be able to have access to a document which lists all/any open-source dependencies of monaco (and its creation process) and their respective licenses. Currenly this is not available.

**Describe the solution you'd like**
Keptn has a nice approach of using a separate github workflow to generate the bill of materials: https://github.com/keptn/keptn/actions/runs/1416203195/workflow. We should follow the same approach with monaco.

### PR DESCRIPTION
**Pull request by didiladi**
This commit adds a workflow for generating the bill of materials for
`monaco` and its dependencies (including the documentation). The
workflow needs to be started manually for now. At a later point in time,
we can run it for every release. But for now: let's keep it simple.

The approach is "borrowed" from `keptn/keptn` :)

fixes #466

### CODE REVIEW

**Comment by didiladi** [comment]
This is a successful run of the workflow: https://github.com/didiladi/dynatrace-monitoring-as-code/actions/runs/1425858305

Go output:
```
github.com/golang/mock,v1.6.0,Apache-2.0
github.com/google/go-cmp,v0.5.6,BSD-3-Clause
github.com/google/uuid,v1.3.0,BSD-3-Clause
github.com/jcelliott/lumber,v0.0.0-20160324203708-dd349441af25,MIT
github.com/pkg/errors,v0.8.1,BSD-2-Clause
github.com/spf13/afero,v1.6.0,Apache-2.0
github.com/urfave/cli/v2,v2.3.0,MIT
gopkg.in/yaml.v2,v2.4.0,Apache-2.0
gotest.tools,v2.2.0+incompatible,Apache-2.0
```

Node output:
```
kessler,stuff,@cmfcmf/docusaurus-search-local,perpetual,material,MIT,https://github.com/cmfcmf/docusaurus-search-local.git,0.6.7,0.6.2,Christian Flach
kessler,stuff,@docusaurus/core,perpetual,material,MIT,git+https://github.com/facebook/docusaurus.git,2.0.0-alpha.75,2.0.0-alpha.75,n/a
kessler,stuff,@docusaurus/preset-classic,perpetual,material,MIT,git+https://github.com/facebook/docusaurus.git,2.0.0-alpha.75,2.0.0-alpha.75,n/a
kessler,stuff,@mdx-js/react,perpetual,material,MIT,git+https://github.com/mdx-js/mdx.git,1.6.22,1.6.21,John Otander
kessler,stuff,@svgr/webpack,perpetual,material,MIT,https://github.com/gregberge/svgr/tree/master/packages/webpack,5.5.0,5.5.0,Greg Bergé
kessler,stuff,clsx,perpetual,material,MIT,git+https://github.com/lukeed/clsx.git,1.1.1,1.1.1,Luke Edwards
kessler,stuff,file-loader,perpetual,material,MIT,git+https://github.com/webpack-contrib/file-loader.git,6.2.0,6.2.0,Tobias Koppers @sokra
kessler,stuff,react,perpetual,material,MIT,git+https://github.com/facebook/react.git,17.0.2,17.0.1,n/a
kessler,stuff,react-dom,perpetual,material,MIT,git+https://github.com/facebook/react.git,17.0.2,17.0.1,n/a
kessler,stuff,url-loader,perpetual,material,MIT,git+https://github.com/webpack-contrib/url-loader.git,4.1.1,4.1.1,Tobias Koppers @sokra
```

**Comment by DavidPHirsch** [comment]
@didiladi can we merge this PR so the Bill of materials can be made public?

**Comment by didiladi** [comment]
@DavidPHirsch sure, let's merge it. I wanted to wait for some additional feedback from anyone in @dynatrace-oss/mac-maintainers, but seems like its ok for everyone.

**Review by tatomir146 — APPROVED** [review]

**Comment by oleg-nenashev** [comment]
Thanks for the change! BOMs are really important in the modern software supply chains

**Pull request merged**
